src/openforms/submissions/signals.py

```diff
@@ -57,3 +60,22 @@ def delete_submission_report_files(
     logger.debug("Deleting file %r", instance.content.name)
 
     instance.content.delete(save=False)
+
+
+@receiver(submission_complete, dispatch_uid="submission.increment_form_counter")
+def increment_form_counter(sender, instance: Submission, **kwargs):
+    submitted_form = instance.form
+
+    try:
+        existing_form_statistics = FormStatistics.objects.get(form=submitted_form)
+    except FormStatistics.DoesNotExist:
+        FormStatistics.objects.create(
+            form=submitted_form,
+            form_name=submitted_form.name,
+            submission_count=1,
+            last_submission=timezone.now(),
+        )
```

Comment:
why not `FormStatistics.objects.get_or_create`?

Reply:
See https://github.com/open-formulieren/open-forms/pull/3789#discussion_r1461634744, probably relevant

Reply:
Check, so then let's do the following:

* Change this block into `get_or_create`
* Keep the update with `F` expression as is, but add a TODO comment specifying that `update_or_create` should be used when we upgrade to Django 5+ (it'll be Django 5.2, we only use LTS releases)